### PR TITLE
Details page redirection

### PR DESCRIPTION
**Pull request by Eclipse-Dominator**
### Summary:
Partially addresses #69, depends on #129 

Adds basic redirection to the profile page via clicking on the user icon
Currently, issues reset during the initialization of `issue viewer`, we should only initialize it upon repo change to cache old data and not reload every time the user navigates back to the dashboard.

### Changes Made:
- Adds profile page routing to the main page
- Moves page resets to upon repo change
- Adds routing to the profile page
- Enables users to access the profile page by clicking on their profile

### Commit Message:

```
We need to access a new page for the profile page.
Currently, issues reset during the initialization of `issue viewer`, we should only initialize it upon repo change to cache old data and not reload every time the user navigates back to the dashboard.
Let's 
- Adds profile page routing to the main page
- Moves page resets to upon repo change
- Adds routing to the profile page
- Enables users to access the profile page by clicking on their profile
```

### CODE REVIEW

File: src/app/detailed-viewer/detailed-viewer.component.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    private logger: LoggingService
+  ) {
+    this.repoChangeSubscription = this.phaseService.repoChanged$.subscribe((newRepo) => {
+      this.initialize();
+    });
+  }
+
+  ngOnInit() {
+    if (this.route.snapshot.paramMap.get('name') === undefined) {
+      this.logger.info('detailed-viewer: Missing username');
```
Comment (chunweii) [inline src/app/detailed-viewer/detailed-viewer.component.ts:45]:
```suggestion
      this.logger.info('DetailedViewerComponent: Missing username');
```

File: src/app/detailed-viewer/detailed-viewer.component.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  styleUrls: ['./detailed-viewer.component.css']
+})
+export class DetailedViewerComponent implements OnInit, OnDestroy, AfterViewInit {
+  public user: GithubUser = undefined;
+
+  private userSubscription: Subscription;
+
+  /** Observes for any change in repo*/
+  repoChangeSubscription: Subscription;
+  issueChangeSubscription: Subscription;
```
Comment (chunweii) [inline src/app/detailed-viewer/detailed-viewer.component.ts:28]: This variable seems unused.

File: src/app/detailed-viewer/detailed-viewer.component.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    private router: Router,
+    private logger: LoggingService
+  ) {
+    this.repoChangeSubscription = this.phaseService.repoChanged$.subscribe((newRepo) => {
+      this.initialize();
+    });
+  }
+
+  ngOnInit() {
+    if (this.route.snapshot.paramMap.get('name') === undefined) {
```
Comment (chunweii) [inline src/app/detailed-viewer/detailed-viewer.component.ts:44]: The return result could be null, which is different from undefined.

Perhaps we can use the falsy value

File: src/app/issues-viewer/card-view/card-view.component.ts [hunk 4]
```diff
@@ -50,4 +52,12 @@ export class CardViewComponent implements OnInit, AfterViewInit, OnDestroy, Filt
   retrieveFilterable(): FilterableSource {
     return this.issues;
   }
+
+  viewUserInBrowser() {
```
Comment (chunweii) [inline src/app/issues-viewer/card-view/card-view.component.ts:56]: Can we rename this to `switchToUserDetailPhase` or something similar?

I was confused by this method initially. I thought this method will open the user's github page in a new tab or browser window.

**Review by chunweii — CHANGES_REQUESTED** [review]
`DetailedViewer` sounds vague. Perhaps we can change the module and component name to `UserDetailViewer`.

Also, you should add a new phase `UserDetailPhase`.

File: src/app/detailed-viewer/detailed-viewer.component.ts (hunk 3 above)
Comment (Eclipse-Dominator) [reply src/app/detailed-viewer/detailed-viewer.component.ts]: Yes, thanks for pointing out, I misread the return type of paremMap.get()
Luckily a separate check was done before this so this is actually never reached, it's more like defensive programming here.

**Comment by Eclipse-Dominator** [comment]
I have added the new phase as well as setting phase values in the constructor of each page to account for forward and backward navigation of pages

**Review by chunweii — APPROVED** [review]

**Pull request merged**